controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -775,11 +787,34 @@ public CompletableFuture<CreateStreamStatus.Status> createStream(String scope, S
                                                                      long createTimestamp, long requestId) {
         log.debug(requestId, "createStream with resource called.");
         OperationContext context = streamMetadataStore.createStreamContext(scope, stream, requestId);
-
-            return execute(
+        if (config.getRetentionPolicy() == null && this.globalRetentionPolicy.get()) {
+            config = createRetentionPolicy(config);
+        }
+        final StreamConfiguration  streamConfig = config;
+        return execute(
```

Comment:
New streams with no retention policy will be applying the global one, but what is the expected behavior for legacy streams in the system? If we want to update multiple stream configuration form existing streams, maybe we can add an Admin Tool command for it (that goes through the scopes, gets the streams, and applies to them the right retention policy).